### ISSUE TITLE
Atom-Typescript error: Maximum call stack size exceeded

### ISSUE THREAD

**Issue by imoverclocked** [atom-haskell/ide-haskell#230]
1. download/unpack https://hackage.haskell.org/package/gnuplot-0.5.6.1/gnuplot-0.5.6.1.tar.gz
2. Add directory as project folder
3. Choose cabal-v2 as build type when prompted
4. peruse source.
5. Bug was triggered by viewing src/Graphics/GnuPlot/Plot/ThreeDimensional.hs

**Atom**: 1.56.0 x64 (on arm64e)
**Electron**: 9.4.4
**OS**: macOS 11.2.3
**Thrown From**: [ide-haskell](https://github.com/atom-haskell/ide-haskell) package 2.6.0

### Stack Trace

Atom-Typescript error: Maximum call stack size exceeded

```
At RangeError: Maximum call stack size exceeded

RangeError: Maximum call stack size exceeded
    at k.filter (/packages/ide-haskell/dist/main.js:1:59942)
    at filter.next (<anonymous>)
    at Function.from (<anonymous>)
    at bd.updateItems (/packages/ide-haskell/dist/main.js:1:186823)
    at /packages/ide-haskell/dist/main.js:1:184221
    at Function.simpleDispatch (/Users/me/Applications/Atom.app/Contents/Resources/app/static/<embedded>:11:1182710)
    at Emitter.emit (/Users/me/Applications/Atom.app/Contents/Resources/app/static/<embedded>:11:1184151)
    at k.didUpdate (/packages/ide-haskell/dist/main.js:1:59751)
    at ko.setMessages (/packages/ide-haskell/dist/main.js:1:59110)
    at Object.setMessages (/packages/ide-haskell/dist/main.js:1:172151)
    at UPIConsumer.setMessages (/packages/ide-haskell-repl/lib/upiConsumer.js:52:22)
    at IdeHaskellReplBg.setErrors (/packages/ide-haskell-repl/lib/ide-haskell-repl-base.js:391:26)
    at IdeHaskellReplBg.clearErrors (/packages/ide-haskell-repl/lib/ide-haskell-repl-base.js:221:14)
    at IdeHaskellReplBg.destroy (/packages/ide-haskell-repl/lib/ide-haskell-repl-base.js:234:14)
    at IdeHaskellReplBg.destroy (/packages/ide-haskell-repl/lib/ide-haskell-repl-bg.js:26:22)
    at Object.dispose (/packages/ide-haskell-repl/lib/upiConsumer.js:125:26)
    at /packages/ide-haskell-repl/lib/upiConsumer.js:132:19
    at Function.simpleDispatch (/Users/me/Applications/Atom.app/Contents/Resources/app/static/<embedded>:11:1182710)
    at Emitter.emit (/Users/me/Applications/Atom.app/Contents/Resources/app/static/<embedded>:11:1184151)
    at IdeHaskellReplBg.destroy (/packages/ide-haskell-repl/lib/ide-haskell-repl-base.js:233:22)
    at IdeHaskellReplBg.destroy (/packages/ide-haskell-repl/lib/ide-haskell-repl-bg.js:26:22)
    at Object.dispose (/packages/ide-haskell-repl/lib/upiConsumer.js:125:26)
    at /packages/ide-haskell-repl/lib/upiConsumer.js:132:19
    at Function.simpleDispatch (/Users/me/Applications/Atom.app/Contents/Resources/app/static/<embedded>:11:1182710)
    at Emitter.emit (/Users/me/Applications/Atom.app/Contents/Resources/app/static/<embedded>:11:1184151)
    at IdeHaskellReplBg.destroy (/packages/ide-haskell-repl/lib/ide-haskell-repl-base.js:233:22)
    at IdeHaskellReplBg.destroy (/packages/ide-haskell-repl/lib/ide-haskell-repl-bg.js:26:22)
    at Object.dispose (/packages/ide-haskell-repl/lib/upiConsumer.js:125:26)
    at /packages/ide-haskell-repl/lib/upiConsumer.js:132:19
    at Function.simpleDispatch (/Users/me/Applications/Atom.app/Contents/Resources/app/static/<embedded>:11:1182710)
```

### Commands

```
```

### Non-Core Packages

```
atom-ide-ui 0.13.0 
env-from-shell 0.3.0 
ide-golang 0.0.5 
ide-gopls 0.2.2 
ide-haskell 2.6.0 
ide-haskell-cabal 2.5.0 
ide-haskell-hasktags 0.0.17 
ide-haskell-hlint 0.1.2 
ide-haskell-hoogle 0.1.3 
ide-haskell-repl 0.10.0 
language-haskell 1.22.0 
```

**Comment by lierdakil** [comment]
I couldn't reproduce the issue. However, I tried rewriting the code that as far as I can tell threw the error, which might be enough. See if the issue reproduces in 2.7.0

**Comment by lierdakil** [comment]
Scratch that. I did evenually reproduce the issue, and tracked it down to ide-haskell-repl. Should be fixed by ide-haskell-repl v0.10.1. Let me know if not.

**Comment by imoverclocked** [comment]
Looks good! Thanks for the quick fix :)
